x-pack/winlogbeat/module/powershell/_meta/fields.yml

```diff
@@ -127,6 +127,14 @@
 
         - name: script_block_text
           type: text
+          analyzer:
+            winlogbeat_powershell_script_analyzer:
+              type: pattern
+              pattern: "[\\W&&[^-]]+"
+          search_analyzer:
+            winlogbeat_powershell_script_analyzer:
+              type: pattern
+              pattern: "[\\W&&[^-]]+"
```

Comment:
Just a note. I duplicated the definition in the search analyzer although this is not required since the analyzers are collated during the fields walk. It is included for clarity — though I'm not sure it achieves that.